model/runner/ProctoringRunnerService.php

```diff
@@ -39,30 +40,31 @@ class ProctoringRunnerService extends QtiRunnerService
     use OntologyAwareTrait;
 
     /**
-     * Get Test Context.
+     * Get Test Data.
      *
      * @param RunnerServiceContext $context
      * @return array
      * @throws \common_Exception
      * @throws \common_exception_NotFound
      * @throws \core_kernel_persistence_Exception
      */
-    public function getTestContext(RunnerServiceContext $context)
+    public function getTestData(RunnerServiceContext $context)
     {
-        $response = parent::getTestContext($context);
+        $couldBePaused = $this->getServiceManager()->get(SectionPauseService::SERVICE_ID)->couldBePaused($context->getTestSession());
+        $testData = parent::getTestData($context);
 
-        if (isset($response['options']) && isset($response['options']['sectionPause'])) {
-            $response['securePauseStateRequired'] = $response['options']['sectionPause'];
+        if (isset($couldBePaused)) {
```

Comment:
I'm having a doubt. 
May be this `if` was only there for backward compatibility and then I don't see how we will fall into the `else` since `canBePaused` will always have a value. 
@handleman could you please verify ; 
 - in tao community when are we in one case and when or we in the other
 - in tao with proctoring (act for example) when are we in one case and when or we in the other

I'm wondering if this condition was only there for backward compatibllity

Reply:
After discussion with @alexeyvy I decided to revert changes to previous state where 
`if (isset($testContext['options']) && isset($testContext['options']['sectionPause']))`

presence of this parameter is checked instead of its value. and now it makes sense to check with `isset`

Reply:
now `canBePaused` is checked this way: `if ($couldBePaused !== null) {`